payara-micro-gradle-plugin/src/main/java/fish/payara/gradle/plugins/micro/PayaraMicroExtension.java

```diff
@@ -78,64 +92,84 @@ public PayaraMicroExtension(Project project) {
         this.project = project;
     }
 
-    public Boolean getSkip() {
-        return skip;
+    public Project getProject() {
+        return project;
+    }
+
+    public boolean isSkip() {
+        return Boolean.valueOf(System.getProperty(PLUGIN_ID + ".skip", String.valueOf(skip)));
     }
 
-    public void setSkip(Boolean skip) {
+    public void setSkip(boolean skip) {
         this.skip = skip;
     }
 
-    public Boolean getImmediateExit() {
-        return immediateExit;
+    public boolean isImmediateExit() {
+        return Boolean.valueOf(System.getProperty(PLUGIN_ID + ".immediateExit", String.valueOf(immediateExit)));
     }
 
-    public void setImmediateExit(Boolean immediateExit) {
+    public void setImmediateExit(boolean immediateExit) {
         this.immediateExit = immediateExit;
     }
 
-    public Boolean getDaemon() {
-        return daemon;
+    public boolean isDaemon() {
+        return Boolean.valueOf(System.getProperty(PLUGIN_ID + ".daemon", String.valueOf(daemon)));
     }
 
-    public void setDaemon(Boolean daemon) {
+    public void setDaemon(boolean daemon) {
         this.daemon = daemon;
     }
 
     public String getJavaPath() {
-        return javaPath;
+        return System.getProperty(PLUGIN_ID + ".javaPath", javaPath);
     }
 
     public void setJavaPath(String javaPath) {
         this.javaPath = javaPath;
     }
 
-    public Boolean getUseUberJar() {
-        return useUberJar;
+    public boolean isUseUberJar() {
+        return Boolean.valueOf(System.getProperty(PLUGIN_ID + ".useUberJar", String.valueOf(useUberJar)));
```

Comment:
I would extract the getting of system property with specific format to an encapsulated method. It's quite repetitive. adding the PLUGIN_ID every time. Not necessary to pass review. Just an observation